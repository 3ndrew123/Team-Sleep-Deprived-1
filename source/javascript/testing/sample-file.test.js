/**
* @jest-environment jsdom
*/


//Import the sum function from sample file
//Testing sample change
const { BrowserRouter } = require('react-router-dom');
var sum = require('../sample-file.js');

//Syntax for testing:
//test("sample test name", test-function())
test('100 + 200 = 300', function(){
	//Syntax for expected output: expect(actualVal).toBe(expectedVal);
	expect(sum(100,200)).toBe(300);
});

/**
 * Sample abstracted function to test sum between any two integers
 * @param {int} a 
 * @param {int} b 
 */
function testAdd(a, b){
	test(`${a} + ${b} = ${a+b}`, function(){
		expect(sum(a,b)).toBe(a+b);
	});
}

//Testing the sum of all integers between -3 and 3
for(let i = -4; i < 4; i++){
	for(let j = -4; j < 4; j++){
		testAdd(i, j);
	}
}

//testing to see if any of the html inside a specific block has changed
test('Test sample snapshot', () => {
	const html = `
	<div>
        <form id='box'>

            <div class="inputsec">
                <label for="username" class="txt">Username: </label><br>
                <input type="text" id="username" class= "txtinput" name="username"><br>
            </div>

            <div class="inputsec" id="passwordgroup">
                <label for="password" class="txt">Password: </label><br>
                <input type="password" id="password" class= "txtinput" name="password"><br>
            </div>

            <div id="check">
                <input type="checkbox" id="rememberme" name="rememberme" value="rememberme">
                <label for="rememberme" class="txt">Remember Me</label><br>
            </div>
        
            <input type="submit" id='submit'  value="Login >">
        </form>
    </div>  
	`
	expect(html).toMatchSnapshot()
})
