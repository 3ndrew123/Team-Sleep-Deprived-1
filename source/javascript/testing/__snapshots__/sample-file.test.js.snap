// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom component 1`] = `undefined`;

exports[`Test sample snapshot 1`] = `
"
	<div>
        <form id='box'>

            <div class="inputsec">
                <label for="username" class="txt">Username: </label><br>
                <input type="text" id="username" class= "txtinput" name="username"><br>
            </div>

            <div class="inputsec" id="passwordgroup">
                <label for="password" class="txt">Password: </label><br>
                <input type="password" id="password" class= "txtinput" name="password"><br>
            </div>

            <div id="check">
                <input type="checkbox" id="rememberme" name="rememberme" value="rememberme">
                <label for="rememberme" class="txt">Remember Me</label><br>
            </div>
        
            <input type="submit" id='submit'  value="Login >">
        </form>
    </div>  
	"
`;
